Feat: add static serving

JedBeom opened the pull request:
## Summary

- Add static serving for `/`

## Test Plan

- Build expo web

```bash
cd frontend
pnpm expo export --platform web
```

- Edit `.env`

```bash
SERVE_DIR=frontend/dist
```

- Start server

```bash
cargo run
```

- Access webpage and see the result

chocological00 commented on src/main.rs:
```diff
@@ -88,12 +89,28 @@ async fn main() -> Result<()> {
     // setup database
     let pool = db::init(&env::var("DATABASE_URL")?).await?;
 
+    let default_serve_dir = "frontend/dist";
```
make this a `const` outside of `fn main`

chocological00 commented on src/main.rs:
```diff
@@ -88,12 +89,28 @@ async fn main() -> Result<()> {
+    let env_serve_dir =
+        env::var("SERVE_DIR").unwrap_or(default_serve_dir.to_owned());
+
+    let path = std::path::Path::new(&env_serve_dir);
```
nit: do `use std::path::Path` on top of file and just call `Path::new()`

chocological00 commented on src/main.rs:
```diff
@@ -88,12 +89,28 @@ async fn main() -> Result<()> {
+
+    // get dist file from env
+    let env_serve_dir =
+        env::var("SERVE_DIR").unwrap_or(default_serve_dir.to_owned());
```
do this:
```rust
let env_serve_dir = env::var("SERVE_DIR").unwrap_or_else(|| default_serve_dir.to_string());
```

chocological00 commented on src/main.rs:
```diff
@@ -88,12 +89,28 @@ async fn main() -> Result<()> {
+
+    let path = std::path::Path::new(&env_serve_dir);
+
+    let serve_dir = if path.exists() && path.is_dir() {
+        path.to_str().unwrap_or(default_serve_dir)
+    } else {
+        default_serve_dir
+    };
+
+    // serve frontend/dist and fallback to index.html
+    let serve_dir = ServeDir::new(serve_dir).not_found_service(ServeFile::new(
+        serve_dir.to_owned() + "/index.html"
+    ));
```
do:
```rust
use color_eyre::eyre::eyre;
---

if !path.is_dir() {
  return Err(eyre!("Path {:#?} is not a valid directory!", &path));
}

let serve_dir = ServeDir::new(path).not_found_service(ServeFile::new(format!("{}/index.html", serve_dir)));

chocological00 commented on src/main.rs:
```diff
@@ -88,12 +89,28 @@ async fn main() -> Result<()> {
-            "/",
-            get(|| async { "Hello, world!" }.instrument(info_span!("/")))
-        )
+        .fallback_service(serve_dir)
```
put this to the bottom right before state

JedBeom committed:
Feat: add static serving

## Summary

- Add static serving for `/`

## Test Plan

- Build expo web

```bash
cd frontend
pnpm expo export --platform web
```

- Edit `.env`

```bash
SERVE_DIR=frontend/dist
```

- Start server

```bash
cargo run
```

- Access webpage and see the result

chocological00 reviewed (approved):
LGTM
